### PR TITLE
Develop ==> Master

### PR DESCRIPTION
**Pull request by smileycreations15**
\+ ~~wut there is a merge conflict~~ (WIP DO NOT MERGE)

### CODE REVIEW

**Comment by smileycreations15** [comment]
@VFDan I fixed it

**Comment by smileycreations15** [comment]
@VFDan Ken still has updates to push.

**Comment by smileycreations15** [comment]
@VFDan Also you can start a copy locally.

**Comment by smileycreations15** [comment]
(if that's what you want)

**Comment by Accio1** [comment]
I made a local copy of the site, and nothing has styles applied. Your links to stylesheets, favicons, etc. are broken.

**Comment by smileycreations15** [comment]
@Accio1 We will start transitioning to the webserver today. Also, I made everything absolute paths.

**Comment by Accio1** [comment]
What is the link to the webserver?

**Comment by smileycreations15** [comment]
> We will **start transitioning** to the webserver today. Also, I made everything absolute paths.

@Accio1 Read the bold text. We didn't start yet but it will be done today in most 4hrs

**Comment by Accio1** [comment]
Okay, I can't review until I can actually see how everything fits together in the website, as trying to review it right now isn't working.

**Comment by smileycreations15** [comment]
We need to merge to see the changes, @Kenny2github will set it up like that, I don't manage the server.

**Comment by Accio1** [comment]
> We need to merge to see the changes, @Kenny2github will set it up like that, I don't manage the server.

Okay, if that's the case then I will approve these changes so that it can be merged, but please be advised that I haven't actually reviewed anything, so if there are errors, I can't catch them.

**Comment by smileycreations15** [comment]
@Accio1 I checked everything and it's fine, the website fits nicely.

**Comment by Accio1** [comment]
> @Accio1 I checked everything and it's fine, the website fits nicely.

How did you check it if the website isn't up and running?

**Comment by smileycreations15** [comment]
I used a local server.

**Comment by smileycreations15** [comment]
(local instance)

**Comment by smileycreations15** [comment]
you could actually run the app if you have python 3 or greater

**Comment by Accio1** [comment]
> I used a local server.

I did too, and it didn't work.

**Comment by smileycreations15** [comment]
@Accio1 local server != opening the files in browser

**Comment by Accio1** [comment]
> @Accio1 local server != opening the files in browser

I know, I used a local server.

**Comment by smileycreations15** [comment]
@VFDan Be sure to use Python 3 and run `pip3 install -r requirements.txt` then `python3 backend`

**Comment by smileycreations15** [comment]
Also be sure to `cd` to the dir

**Comment by VFDan** [comment]
@smileycreations15 I only have Python 3

**Comment by smileycreations15** [comment]
@VFDan then you can remove the `3` suffix

**Comment by smileycreations15** [comment]
@VFDan Also did you got it working?

**Comment by smileycreations15** [comment]
It should not without a config file

**Comment by VFDan** [comment]
@smileycreations15 haven't started testing yet

**Comment by smileycreations15** [comment]
Edit `config.json` in your local copy @VFDan @Accio1
Set the contents to this:
```json
{"port":8888,"hook-secret":"leave-this-blank"}
```

**Comment by smileycreations15** [comment]
(i forgot to tell that)

**Comment by VFDan** [comment]
@Accio1 Ah ok

**Comment by smileycreations15** [comment]
@VFDan did you intended to ping me?

**Comment by VFDan** [comment]
@smileycreations15 Yes, and what location should I `cd` to? `/public`?

**Comment by smileycreations15** [comment]
@VFDan The root of the project run these:
```sh
pip install -r requirements.txt
python backend
```

**Comment by VFDan** [comment]
Ok

**Comment by smileycreations15** [comment]
@VFDan the project root

**Comment by Accio1** [comment]
I don't have Python

**Comment by smileycreations15** [comment]
wait wat

**Comment by smileycreations15** [comment]
:wut:

**Comment by Accio1** [comment]
> wait wat

I don't have python, and I don't know how to use it.

**Comment by smileycreations15** [comment]
:(
wat

**Comment by Accio1** [comment]
> :(

But I do have a local server (I think).

**Comment by smileycreations15** [comment]
@Accio1 but the server is python :/

**Comment by smileycreations15** [comment]
*is written in python

**Comment by VFDan** [comment]
When I do `python backend` the thing just blinks on the next line (without the starting `C:\...>`).

**Comment by smileycreations15** [comment]
@VFDan yes wait

**Comment by smileycreations15** [comment]
Open localhost:8888

**Comment by Accio1** [comment]
> @Accio1 but the server is python :/

I'm very confused, I think I'll just wait until the regular website is set up, and then review that.

**Comment by VFDan** [comment]
I get a 404

**Comment by smileycreations15** [comment]
@VFDan New url?

**Comment by VFDan** [comment]
@smileycreations15 wdym

**Comment by smileycreations15** [comment]
@VFDan try this: http://localhost:888/site/

**Comment by smileycreations15** [comment]
+ I am pushing 3 updates

**Comment by VFDan** [comment]
@smileycreations15 Yes that works (needed to add an extra 8)

**Comment by smileycreations15** [comment]
@VFDan Can you pull the updates and try again? It should be all done now

**Comment by smileycreations15** [comment]
@VFDan It is a local config. Ken stores the webhook secret required to auto-update the app there

**Comment by smileycreations15** [comment]
(or a user who hosts the server)

**Comment by VFDan** [comment]
@smileycreations15 Sry

File: public/login/index.html [hunk 1]
```diff
@@ -2,38 +2,110 @@
...
+
+    <span>Comment this code on your profile:</span><br>
+
+    <form onsubmit="event.preventDefault();step2()">
+      <input id="code" disabled type="text" placeholder="Loading code..." autocomplete="off" spellcheck="false">
+      <button type="submit" class="smilejs-ripple submit">Continue</button>
+    </form>
+  </div>
+</div>
+
```
[the remark this answers is not in the document]
Comment (smileycreations15) [reply public/login/index.html]: It is not required. You need to create an application after logging in which is the main process of signup

**Comment by Accio1** [comment]
Can I use a python emulator like https://repl.it/languages/python3 ?

**Comment by smileycreations15** [comment]
@Accio1 I will try to get it on there :)

**Comment by VFDan** [comment]
@smileycreations15 I've already gave my review mate, https://github.com/ScratchVerifier/ScratchVerifier/pull/21#discussion_r318811182

**Comment by smileycreations15** [comment]
@VFDan It actually creates an account when logging in if you don't have one

**Comment by VFDan** [comment]
@smileycreations15 Ah okay

**Comment by smileycreations15** [comment]
> It is not required. You need to create an application after logging in which is the main process of signup

**Comment by smileycreations15** [comment]
ok

**Comment by VFDan** [comment]
Nm ^^, I misinterpreted the login page.

**Comment by VFDan** [comment]
I'm going to add the credits before I accept the pull

**Comment by VFDan** [comment]
@smileycreations15 The coders are you, Ken, me, and Accio, right!

**Comment by smileycreations15** [comment]
@VFDan yes.

**Comment by smileycreations15** [comment]
@Accio1 Sure! I recommend http://beautifier.io

**Comment by smileycreations15** [comment]
@VFDan The logo serves the purpose

**Comment by smileycreations15** [comment]
@Accio1 Well yes

**Comment by smileycreations15** [comment]
*spaces*

**Comment by smileycreations15** [comment]
@VFDan Maybe name it "homepage?"

**Comment by smileycreations15** [comment]
@VFDan yes

**Comment by smileycreations15** [comment]
@VFDan Did you got it working?

**Comment by VFDan** [comment]
@smileycreations15 Testing now

**Comment by smileycreations15** [comment]
(it should with the recent change also you should be able to sign up if you pulled the last one)

**Comment by smileycreations15** [comment]
(also pull it!)

**Comment by VFDan** [comment]
@smileycreations15 Pull what?

**Comment by smileycreations15** [comment]
@VFDan the recent changes (pull the changes from the develop branch)

**Comment by smileycreations15** [comment]
(i pushed some updates that you might have not pulled)

**Comment by VFDan** [comment]
@smileycreations15 I have the most recent version in the local copy

**Comment by smileycreations15** [comment]
@VFDan k

**Comment by VFDan** [comment]
Maybe we can add a `Copy` button for the code?

**Comment by smileycreations15** [comment]
@VFDan on it.

**Comment by Accio1** [comment]
I have code for a copy to clipboard button.

**Comment by VFDan** [comment]
@Accio1 Please add that then

**Comment by smileycreations15** [comment]
@Accio1 I can do it with the newest API's

**Comment by Accio1** [comment]
What will our domain be. I need it for the reCAPTCHA

**Comment by smileycreations15** [comment]
http://scratchverifier.ddns.net:8888

File: public/login/index.html [hunk 2]
```diff
@@ -2,38 +2,110 @@
...
+  <img src="https://u.cubeupload.com/smileycreations15/V732dJ.png" width="50%">
+  <br>
+  <h1>Login</h1><br>
+  <div class="Login_Box">
+    <span id="err2" style="color:red;"></span><br>
+
+    <span>Comment this code on your profile:</span><br>
+
+    <form onsubmit="event.preventDefault();step2()">
+      <input id="code" disabled type="text" placeholder="Loading code..." autocomplete="off" spellcheck="false">
```
[the remark this answers is not in the document]
Comment (smileycreations15) [reply public/login/index.html]: On it...

**Comment by Accio1** [comment]
> http://scratchverifier.ddns.net:8888

Do I keep the :8888 on the end?

**Comment by smileycreations15** [comment]
yes

**Comment by Accio1** [comment]
That seems like a weird url

**Comment by smileycreations15** [comment]
@Accio1 Port number.

**Comment by Accio1** [comment]
> @Accio1 Port number.

?

**Comment by smileycreations15** [comment]
@Accio1 the port number of the server

**Comment by Accio1** [comment]
So when people want to find the site they will have to type http://scratchverifier.ddns.net:8888 into the URL bar?

**Comment by smileycreations15** [comment]
yes but we will find a way

**Comment by Accio1** [comment]
> yes but we will find a way

What do you mean?

**Comment by smileycreations15** [comment]
@Accio1 a way to keep the url short

**Comment by Accio1** [comment]
> @Accio1 a way to keep the url short

Why can't you use a different url without the :8888 on the end?

**Comment by smileycreations15** [comment]
@Accio1 I cannot

**Comment by Accio1** [comment]
> @Accio1 I cannot

The Scratch Team doesn't like shortened links, so we won't be able to post the link on Scratch.

**Comment by smileycreations15** [comment]
@Accio1 Frame it in another website

**Comment by Accio1** [comment]
The reCAPTCHA site says "The following domains are invalid: http://scratchverifier.ddns.net:8888. A valid domain requires a host and must not include any path, port, query or fragment."

**Comment by smileycreations15** [comment]
OH!

**Comment by smileycreations15** [comment]
remove the :8888

**Comment by Accio1** [comment]
So we will need a new URL

**Comment by smileycreations15** [comment]
@Accio1 I can add a iframe to a subdomain to my site

**Comment by Accio1** [comment]
> remove the :8888

That brings you here, which I don't think is out site: http://scratchverifier.ddns.net/test/Main_Page

**Comment by smileycreations15** [comment]
@Accio1 Well yes, but it says no port

**Comment by smileycreations15** [comment]
@Accio1 reCaptcha will work with the :8888

**Comment by Accio1** [comment]
> @Accio1 reCaptcha will work with the :8888

No, when I type in the domain, it doesn't let me continue

**Comment by smileycreations15** [comment]
@Accio1 why?

**Comment by Accio1** [comment]
Screenshot 2019-08-28 at 6 47 58 PM

**Comment by smileycreations15** [comment]
@Accio1 Remove the :8888 when signing up

**Comment by smileycreations15** [comment]
but not when using the site

**Comment by Accio1** [comment]
> @Accio1 Remove the :8888 when signing up

But that isn't our site.

**Comment by smileycreations15** [comment]
we are confused rn @Kenny2github

**Comment by smileycreations15** [comment]
@Kenny2github should handle this

**Comment by Accio1** [comment]
> @Kenny2github should handle this

Yes, please

**Comment by VFDan** [comment]
@Accio1 While we wait for Ken, can you please add the `Copy` button?

**Comment by smileycreations15** [comment]
yes

**Comment by smileycreations15** [comment]
did that

**Comment by VFDan** [comment]
@smileycreations15 where?

**Comment by smileycreations15** [comment]
cd7fdc2

**Comment by Accio1** [comment]
I have to go for now, I'll probably be back in 1.5 hours.

**Comment by VFDan** [comment]
ah ok

**Comment by smileycreations15** [comment]
ok

**Comment by VFDan** [comment]
@Accio1 cya

**Comment by Accio1** [comment]
👋

**Comment by smileycreations15** [comment]
👋

**Comment by VFDan** [comment]
@smileycreations15 That is just the link to the homepage

**Comment by smileycreations15** [comment]
oops wrong commit

**Comment by smileycreations15** [comment]
f3f9d5cb4e328c4029f46cda3f7f96efe5c2715b

**Comment by VFDan** [comment]
Ah okay ty

**Comment by smileycreations15** [comment]
@Accio1 We should not do captcha if it is too much work for Ken

**Comment by smileycreations15** [comment]
*please do not get mad at me i don't understand why everyone is getting mad at me*

**Comment by Accio1** [comment]
> _please do not get mad at me i don't understand why everyone is getting mad at me_

I'm not mad

**Comment by Accio1** [comment]
Could the URL be changed to http://scratchverify.ddns.net/ ? It is available

**Comment by smileycreations15** [comment]
@Accio1 still the ports............. same host.......

**Comment by smileycreations15** [comment]
same issue

**Comment by Accio1** [comment]
Why would it be an issue? You don't need to add the :8888 because no one is using the site already.

**Comment by smileycreations15** [comment]
@Accio1 ports ports ports. The same host will be used so again...

**Comment by Accio1** [comment]
> @Accio1 ports ports ports. The same host will be used so again...

What are the ports for?

**Comment by smileycreations15** [comment]
@Accio1 To select what will receive it (like process A or B)

**Comment by Accio1** [comment]
I don't get it

**Comment by smileycreations15** [comment]
@Accio1 it is hard to explain

**Comment by Accio1** [comment]
So the :8888 has to be on the end no matter what?

**Comment by smileycreations15** [comment]
@Accio1 yes

**Comment by Kenny2github** [comment]
@Accio1 The reason the port is required is because we're going to be hosting this on my RPi which is already using ports 80, 443, and 8080 for other purposes, which means that I had to choose another port, so I chose 8888 arbitrarily

**Comment by smileycreations15** [comment]
@Kenny2github Also should we use reCaptcha?

**Comment by smileycreations15** [comment]
or don't?

**Comment by Accio1** [comment]
> @Accio1 The reason the port is required is because we're going to be hosting this on my RPi which is already using ports 80, 443, and 8080 for other purposes, which means that I had to choose another port, so I chose 8888 arbitrarily

Okay, I just don't really understand what a port is.

**Comment by Accio1** [comment]
So can this be merged now?

**Comment by smileycreations15** [comment]
@Accio1 Will do but still working on updates

**Comment by Kenny2github** [comment]
Note to self: rebase develop before merge

**Comment by VFDan** [comment]
@smileycreations15 What is QA?

**Comment by Kenny2github** [comment]
@VFDan QA is "Quality Assurance" - meaning testing, bugfinding, and general cleanup

**Comment by VFDan** [comment]
@Kenny2github I thought it was something like that, thanks!

**Comment by VFDan** [comment]
> Kenny2github assigned Kenny2github, VFDan, smileycreations15 and Accio1
Wait, so what do you want us to do?

**Comment by Kenny2github** [comment]
Assigning myself and @smileycreations15 for pre-merge changes, @VFDan and @Accio1 for review if they feel like it

**Comment by VFDan** [comment]
Okay, I'll review.

**Comment by smileycreations15** [comment]
@VFDan Thanks for the note :D

**Comment by VFDan** [comment]
@Kenny2github Merge time! You do the honors.

**Comment by smileycreations15** [comment]
🎉

**Comment by VFDan** [comment]
🎉

**Comment by VFDan** [comment]
Yay!!

**Comment by smileycreations15** [comment]
Yay!

**Comment by smileycreations15** [comment]
@VFDan @Kenny2github @Accio1 
Screen Shot 2019-08-29 at 06 09 39

**Comment by VFDan** [comment]
Yay! So what is the link?

**Comment by smileycreations15** [comment]
@VFDan Still setting up

**Comment by VFDan** [comment]
Should we delete the `develop` branch?

**Comment by Accio1** [comment]
http://scratchverifier.ddns.net:8888/site/ isn't working

**Comment by smileycreations15** [comment]
@Accio1 There is some stuff going on...

**Comment by Accio1** [comment]
> @Accio1 There is some stuff going on...

When do you think it will be resolved and the site will be up and running?

**Comment by smileycreations15** [comment]
@Accio1 3 days

**Comment by Kenny2github** [comment]
By the time the milestone deadline for "Initial release" has passed, it will be up.

**Pull request merged**